tests/jerry/es2015/typedarray-from.js

```diff
@@ -0,0 +1,162 @@
+var weak_set = new WeakSet();
+weak_set.add(foo);
+weak_set.add(bar);
+
+var string = new String('123');
+
+var map = new Map();
+map.set(0, 'zero');
+map.set(1, 'one');
+
+var weak_map = new WeakMap();
+weak_map.set(foo, 'foo');
+weak_map.set(bar, 'bar');
+
+var string = '123';
+
+for (var constructor of typedArrayConstructors)
+{
+    assert(constructor.from.length === 1);
+
+    try {
+      function f() {constructor.from.call(Array, []);}
+    } catch (e) {
+      assert(e instanceof TypeError); 
+    }
+
+    assert(constructor.from(set).toString() === '1,2,3');
+    assert(constructor.from(weak_set).toString() === '');
+
+    if (constructor == Float32Array || constructor == Float64Array)
+    {
+        assert(constructor.from(map).toString() === 'NaN,NaN');
+    }
+    else
+    {
+        assert(constructor.from(map).toString() === '0,0');
+    }
+
+    assert(constructor.from(weak_map).toString() === '');
+    assert(constructor.from(string).toString() === '1,2,3');
+
+    try {
+      function f() {constructor.from.call({}, []);}
+    } catch (e) {
+      assert(e instanceof TypeError);
+    }
+
+    try {
+      function f() {constructor.from.call([], []);}
+    } catch (e) {
+      assert(e instanceof TypeError);
+    }
+
+    try {
+      function f() {constructor.from.call(1, []);}
+    } catch (e) {
+      assert(e instanceof TypeError);
+    }
+
+    try {
+      function f() {constructor.from.call(undefined, []);}
+    } catch (e) {
+      assert(e instanceof TypeError);
+    }
+
+    assert(constructor.from([1,2,3,4]).toString() === '1,2,3,4');
+    assert(constructor.from([12,45]).toString() === '12,45');
+    assert(constructor.from(NaN).toString() === '');
+    assert(constructor.from(Infinity).toString() === '');
+
+    try {
+      function f() {constructor.from(null)}
+    } catch (e) {
+      assert(e instanceof TypeError);
+    }
+
+    try {
+      function f() {constructor.from(undefined);}
+    } catch (e) {
+      assert(e instanceof TypeError);
+    }
+
+    var called = 0;
+    var arr = [1,2,3];
+    var obj = {};
+
+    function testIterator() {
+      called++;
+      return arr[Symbol.iterator]();
+    }
+
+    var getCalled = 0;
+    Object.defineProperty(obj, Symbol.iterator, {
+      get: function() {
+          getCalled++;
+          return testIterator;
+      },
+    });
+
+    assert(constructor.from(obj).toString() === '1,2,3');
+    assert(called === 1);
+    assert(getCalled === 1);
+}
+
+class SimpleClass {
+    constructor(data) {
+        this.index = 0;
+        this.data = data;
+    }
+
+    [Symbol.iterator]() {
+        return {
+            next: () => {
+                if (this.index < this.data.length) {
+                    return {value: this.data[this.index++], done: false}
+                } else {
+                    this.index = 0;
+                    return {done: true}
+                }
+            }
+        }
+    }
+}
+
+var c = new SimpleClass([1,2,3,4,5]);
```

Comment:
Maybe I'm missing something, but how is this testcase related to the typeaddary.from?